gdal/ogr/ogrsf_frmts/flatgeobuf/ogr_flatgeobuf.h

```diff
@@ -0,0 +1,173 @@
+/******************************************************************************
+ *
+ * Project:  FlatGeobuf driver
+ * Purpose:  Definition of classes for OGR FlatGeobuf driver.
+ * Author:   Björn Harrtell <bjorn at wololo dot org>
+ *
+ ******************************************************************************
+ * Copyright (c) 2018, Björn Harrtell <bjorn at wololo dot org>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ ****************************************************************************/
+
+#ifndef OGR_FLATGEOBUF_H_INCLUDED
+#define OGR_FLATGEOBUF_H_INCLUDED
+
+#include "ogrsf_frmts.h"
+#include "ogr_p.h"
+
+#include "header_generated.h"
+#include "feature_generated.h"
+#include "packedrtree.h"
+
+using namespace FlatGeobuf;
+
+class OGRFlatGeobufDataset;
+
+static constexpr const uint8_t magicbytes[8] = { 0x66, 0x67, 0x62, 0x00, 0x66, 0x67, 0x62, 0x00 };
+
+struct FeatureItem : Item {
+    flatbuffers::DetachedBuffer buf;
+    uint8_t *data;
+    size_t size;
+};
+
+struct GeometryContext {
+    std::vector<double> xy;
+    std::vector<double> z;
+    std::vector<double> m;
+    std::vector<uint32_t> ends;
+    std::vector<uint32_t> lengths;
+};
+
+class OGRFlatGeobufLayer final : public OGRLayer
+{
+    private:
+        std::string m_osFilename;
+        std::string m_osLayerName;
+
+        VSILFILE *m_poFp = nullptr;
+
+        const Header *m_poHeader = nullptr;
+        OGRwkbGeometryType m_eGType;
+        GeometryType m_geometryType;
+        bool m_hasM = false;
+        bool m_hasZ = false;
+        bool m_hasT = false;
+        bool m_hasTM = false;
```

Comment:
what means m_hasTM ?

Reply:
TM stands for "time measurement" introduced by the discussion at https://github.com/bjornharrtell/flatgeobuf/issues/6#issuecomment-515709904. Note that T is for geodetic decimal year time, also discussed on that same issue. May be a futile attempt to support something not yet standardized. :)

Reply:
OK, when compared to https://lists.osgeo.org/pipermail/proj/2019-August/008806.html, I guess  T would be the "Epoch of Expression" , and TM the "Epoch of Observation". In any case adding somewhere a comment to explain the semantics wouldn't hurt.

Reply:
Seems right yes. It's somewhat documented uptream at https://github.com/bjornharrtell/flatgeobuf/blob/master/src/fbs/feature.fbs but deserve something more public if this becomes useful.